### PR TITLE
feat(pipes-enrichments): add Step Functions enrichment eventbridge pipes

### LINKED ISSUE #29385: (pipes-enrichments): Add Step Functions state machine enrichment for EventBridge pipes
### Describe the feature

Add a implementation for the Step Functions state machine enrichment for EventBridge pipes in the `pipes-enrichment` module.

### Use Case

For feature completeness of the L2 pipes module this source is required.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

-

### Environment details (OS name and version, etc.)

-

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

Closes #29385.

### Reason for this change
To use Step Functions state machine enrichment for eventbrige pipes

### Description of changes
Add `StepFunctionsEnrichment` class.

### Description of how you validated changes
Add unit test and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-pipes-enrichments-alpha/README.md [hunk 1]
```diff
@@ -46,3 +46,24 @@ const pipe = new pipes.Pipe(this, 'Pipe', {
   target: new SomeTarget(targetQueue),
 });
 ```
+
+### Step Functions statemachine
+
+A Step Functions statemachine can be used to enrich events of a pipe.
+Express workflows are only supported and can only be invoked synchronously.
+For more information, see [Amazon EventBridge Pipes event enrichment](https://docs.aws.amazon.com/eventbridge/latest/userguide/pipes-enrichment.html).
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-pipes-enrichments-alpha/README.md:54]:
```suggestion
### Step Functions state machine

Step Functions state machine can be used to enrich events of a pipe.

**Note:** EventBridge Pipes only supports Express workflows invoked synchronously.

> Visit [Amazon EventBridge Pipes event enrichment](https://docs.aws.amazon.com/eventbridge/latest/userguide/pipes-enrichment.html) for more details.
```

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍 Left suggestions for minor adjustments

**Comment by mazyu36** [comment]
@lpizzinidev 

Thank you always for your reviews!
I've addressed all your comment.

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

File: packages/@aws-cdk/aws-pipes-enrichments-alpha/lib/stepfunctions.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  bind(pipe: IPipe): EnrichmentParametersConfig {
+    return {
+      enrichmentParameters: {
+        inputTemplate: this.inputTransformation?.bind(pipe).inputTemplate,
+      },
+    };
+  }
+
+  grantInvoke(pipeRole: IRole): void {
+    this.stateMachine.grantStartSyncExecution(pipeRole);
```
Comment (GavinZZ) [inline packages/@aws-cdk/aws-pipes-enrichments-alpha/lib/stepfunctions.ts:43]: To help me understand, can you provide any documentation on why this permission is needed? (Want to make sure we don't grant excessive permission or grant too little permission).

**Comment by mazyu36** [comment]
@GavinZZ 
Thank you for the review.
I've replied to your comment.

File: packages/@aws-cdk/aws-pipes-enrichments-alpha/lib/stepfunctions.ts (hunk 2 above)
Comment (GavinZZ) [reply packages/@aws-cdk/aws-pipes-enrichments-alpha/lib/stepfunctions.ts:43]: Thanks for the clarification. That sounds right to me.

**Comment by mazyu36** [comment]
@Mergifyio update

**Pull request merged**